src/quarto/game/game.go

```diff
@@ -63,44 +55,31 @@ func PlacePieceOnGrid(state State) State {
 }
 
 // ChoosePositionForPiece return coordinates to place the next piece
-func ChoosePositionForPiece(state State) [2]int{
-	coord := ChooseWinningtPositionForPiece(state)
-	if (coord == emptyCoord) {
-		coord = ChooseFirstPositionForPiece(state)
+func ChoosePositionForPiece(state State) [2]int {
+	coord := ChooseWinningPositionForPiece(state)
+	if coord == emptyCoord {
+		coord = ChooseRandomPositionForPiece(state)
 	}
 	return coord
 }
 
-// ChooseWinningtPositionForPiece return first winning coordinates to place the next piece if exists
-func ChooseWinningtPositionForPiece(state State) [2]int{
-	coord := emptyCoord
-	size := GetGridSize(state)
-	for i := 0; i < size; i++ {
-		for j := 0; j < size; j++ {
-			if state.Grid[i][j] == 0 {
-				coord[0] = i
-				coord[1] = j
-				return coord
-			}
+// ChooseWinningPositionForPiece return first winning coordinates to place the next piece if exists
+func ChooseWinningPositionForPiece(state State) [2]int {
+	coordList := grid.GetEmptyBoxes(state.Grid)
+	for i := 0; i < len(coordList); i++ {
+		if grid.IsWinningPosition(coordList[i][0], coordList[i][1], state.Grid, state.Piece) {
+			return coordList[i]
+
 		}
 	}
-	return coord
+	return emptyCoord
```

Comment:
Where come from this variable?

Reply:
I get it. Can you return nil instead of an "emptyCoord" to handle the case when you can't have a winning move ?